### PR TITLE
Golf proofs in `Minimal.lean`

### PR DESCRIPTION
**Pull request by pitmonticone**
Co-Authored-By: Lorenzo Luccioli <71074618+LorenzoLuccioli@users.noreply.github.com> @LorenzoLuccioli

### CODE REVIEW

**Review by oliver-butterley — APPROVED** [review]
Great! Well done @pitmonticone and @LorenzoLuccioli  :rocket: 

Before proceeding any further, we need to decide definitely the way to match this up with what already exists in mathlib and drop the requirement on the space being metrizable. Let's discuss when we meet in Pisa! Alternatively we could organise a call sometime before then.

**Comment by pitmonticone** [comment]
Just to clarify, I will be in Pisa next month, while @LorenzoLuccioli could only be available for a call (if I remember correctly). 

While I am and will be quite busy with my exams (even once in Pisa), I can definitely help with the Mathlib exploration and related design choices.

I will be available next Tuesday from 3 p.m. to 5 p.m.

Let me know if is fine for you all.

File: BET/Minimal.lean [hunk 1]
```diff
@@ -36,27 +36,26 @@ To do: remove invariant, add nonempty. -/
...
 /-- In a minimal dynamics, the recurrent set is all the space. -/
 theorem recurrentSet_of_minimal_is_all_space (hf : IsMinimal f) (x : α) :
     x ∈ recurrentSet f := by
-  have : ∀ (ε : ℝ) (N : ℕ), ε > 0 → ∃ m : ℕ, m ≥ N ∧ f^[m] x ∈ ball x ε := by
-    intro ε N hε
-    obtain ⟨n, hball⟩ : ∃ n, f^[n] (f^[N] x) ∈ ball x ε :=
-      hf.minimal x (f^[N] x) (mem_univ _) (mem_univ _) ε hε
-    exact ⟨n + N, by linarith, (Function.iterate_add_apply _ _ _ _).symm ▸ hball⟩
-  exact (recurrentSet_iff_accumulation_point f x).mpr this
+  apply (recurrentSet_iff_accumulation_point f x).mpr
```
Comment (mseri) [inline BET/Minimal.lean:49]:
```suggestion
  -- explicitly, we are now proving
  -- ∀ (ε : ℝ) (N : ℕ), ε > 0 → ∃ m : ℕ, m ≥ N ∧ f^[m] x ∈ ball x ε
  apply (recurrentSet_iff_accumulation_point f x).mpr
```

File: BET/Minimal.lean [hunk 2]
```diff
@@ -91,102 +90,82 @@ structure IsMinimalAlt (f : α → α) (U : Set α) : Prop :=
...
-    (f : α → α) (C : Set (Set α))
+theorem inter_nested_closed_inv_is_closed_inv_nonempty (f : α → α) (C : Set (Set α))
     (hc1 : C.Nonempty) (hc2 :  IsChain (· ⊆ ·) C) (hn : ∀ V ∈ C, IsCIN f V) :
     IsCIN f (⋂₀ C) := by
   have hScl := (fun V x ↦ (hn V x).closed)
   have hne := (fun V x ↦ (hn V x).nonempty)
+  have htd : DirectedOn (· ⊇ ·) C := IsChain.directedOn hc2.symm
+  have hSc : ∀ U ∈ C, IsCompact U := fun U a ↦ IsClosed.isCompact (hScl U a)
+  have : Nonempty C := nonempty_coe_sort.mpr hc1
+  refine' ⟨_, isClosed_sInter hScl, fun n x hx U h2c ↦ _⟩
```
Comment (mseri) [inline BET/Minimal.lean:106]:
```suggestion
  -- the use of fun in the constructor in the following refine
  -- allows one to introduce directly all the relevant terms
  -- for the proof
  refine' ⟨_, isClosed_sInter hScl, fun n x hx U h2c ↦ _⟩
```

File: BET/Minimal.lean [hunk 3]
```diff
@@ -91,102 +90,82 @@ structure IsMinimalAlt (f : α → α) (U : Set α) : Prop :=
...
-    have h4 : ∀ V ∈ C, IsCIN f V := by
-      intro V h5
-      exact (h1 h5).right
+    have h4 : ∀ V ∈ C, IsCIN f V := fun V h5 ↦ (h1 h5).right
     have h5 := inter_nested_closed_inv_is_closed_inv_nonempty f C h3 h2 h4
     /- We show that `lb` is in `S`. -/
     choose V' h8 using h3 -- Let's fix some `V ∈ C`.
-    have h14 : V' ∈ S := by exact h1 h8
-    have h6 : lb ⊆ U := by exact Subset.trans (sInter_subset_of_mem h8) (h14.left)
+    have h14 : V' ∈ S := h1 h8
```
Comment (mseri) [inline BET/Minimal.lean:129]:
```suggestion
    -- same as have h14 : V' ∈ S := by exact h1 h8
    have h14 : V' ∈ S := h1 h8
```

File: BET/Minimal.lean [hunk 4]
```diff
@@ -91,102 +90,82 @@ structure IsMinimalAlt (f : α → α) (U : Set α) : Prop :=
...
 
 /-- The orbit of a point is invariant. -/
 theorem orbit_inv (x : α) : IsInvariant (fun n x ↦ f^[n] x) (orbit f x) := by
-  intros n y h0
+  intro n y h0
   choose m h1 using h0
-  have h5 : f^[n] y = f^[n + m] x := by
-    rw [h1]
-    exact (iterate_add_apply f n m x).symm
   use n + m
```
Comment (mseri) [inline BET/Minimal.lean:150]:
```suggestion
  -- here we show that f^[n] y = f^[n + m] x
  use n + m
```

**Review by mseri — COMMENTED** [review]
Thanks! I’d like to keep a few comments around, for the sake of learning and legibility. Besides that, for me it is good to merge

**Comment by pitmonticone** [comment]
Committed all the suggested changes.

**Comment by mseri** [comment]
Thanks!

**Comment by mseri** [comment]
Next week I am fully booked the whole week :'O

**Pull request merged**